src/Random/Array.elm

```diff
@@ -89,27 +90,21 @@ choose arr =
             gen
 
 
-{-| Shuffle the array using the Fisher-Yates algorithm. Takes O(_n_ log _n_)
-time and O(_n_) additional space.
+{-| Shuffle the list using the Union-Find data structure with path compression algorithm.
+Takes O(_n_ log _n_).
```

Comment:
Can you be more specific as to whether this is time, space, or both?

Reply:
Unfortunately I'm not a big pro of the question. Be honest I've started to lear algorithms and data structures a couple of weeks ago.

The `UnionFind` creates a `Dict` underhood, size of the dict at the end of shuffle will be equal `n - 1`. I assume that is the additional space, isn't it?

Reply:
Seems reasonable. So `Takes O(_n_ log _n_) time and space.` then.